projects/worker-azure-ai/actions/worker-azure-ocr/index.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
+
+const { worker, GenericError } = require("@adobe/asset-compute-sdk");
+const fetch = require("node-fetch");
+const fs = require('fs').promises;
+
+const DEFAULT_LANGUAGE = "en";
+
+exports.main = worker(async (source, rendition, params) => {
+    // get Azure credentials for API
+    let subscriptionKey = params && params.AZURE_OCP_KEY;
+    let endpoint = params && params.AZURE_OCP_ENDPOINT;
+
+    // WORKER_TEST_MODE is true when you are running worker unit
+    // tests using the `test-worker` command
+    // Dummy authorization parameters needed for mocks 
+    if (process.env.WORKER_TEST_MODE) {
+        subscriptionKey = "test-azure-key";
+        endpoint = "https://westus.api.cognitive.microsoft.com/";
+    }
+
+    // check that credentials are set before calling API
+    if (!subscriptionKey || !endpoint) { 
+        throw new Error("Missing required credentials for Azure OCR Api.");
+    }
+
+    // format and execute API request
+    const language = (rendition.instructions && rendition.instructions.language) || DEFAULT_LANGUAGE;
+    const url = `${endpoint}vision/v3.0/ocr?language=${language}&detectOrientation=true`;
+    const options = {
+        method:"POST",
+        body: JSON.stringify({ 
+            url: source.url
+        }),
+        headers: {
+            "Content-Type": "application/json",
+            "Ocp-Apim-Subscription-Key" : subscriptionKey
+        }
+    };
+
+    const response = await fetch(url, options);
+    const jsonResponse = await response.json();
+    if (jsonResponse.code || jsonResponse.error) {  
```

Comment:
So there is a `jsonResponse.code` only on failure, not on success ('20*')?

Reply:
the code is not a number, it's a string value defined by the azure api: https://westcentralus.dev.cognitive.microsoft.com/docs/services/5adf991815e1060e6355ad44/operations/56f91f2e778daf14a499e1fc
```
{
    "code":"InvalidImageFormat",
    "requestId":"B8D802CF-DD8F-4E61-B15C-9E6C5844CCBC",
    "message":"The input file is not in a valid image format that the service can support. "
}
```